### ISSUE TITLE
Recommend SPDX and CycloneDX for BOM links in `byproducts`

### ISSUE THREAD

**Issue by MarkLodato** [slsa-framework/slsa#691]
Feedback from @mrutkows: it would be valuable to call out explicitly putting a link to a BOM in `byproducts`. This is similar to #669 from @chitrangpatel that it's not clear where to put the build spec.

I'm thinking of guidelines that, if you want to describe more details about the runtime environment, we recommend using a well established format like SPDX or CycloneDX to represent these details.

**Comment by MarkLodato** [comment]
No, at least not for this issue. `resolvedDependencies` (and `buildDefinition` in general) is about the *inputs* to the build. That does overlap with **Software** Bill of Materials, but is not the intention here. If you ran this build on another service, the `buildDefinition` (and most of the SBOM) would remain the same.

Rather, this issue is about describing the `runDetails`, i.e. the specific invocation of this build. The BOM here would be a "Service" Bill of Materials. For example, what version and configuration of Tekton was running on the build cluster.

**Comment by laurentsimon** [comment]
Got it, thanks for the clarification

**Comment by MarkLodato** [comment]
SBOM is intentionally not part of the Build track, though perhaps might be part of a future Vulnerabilities track. We really need a FAQ on this.

My quick take:

- Provenance and the Build track are focused on trustworthiness of the build process. Provenance is necessarily generated in the build platform's trusted control plane, which in practice results in it being coarse grained.
- SBOM is focused on vulnerability management and licensing. In practice, this requires a signal-to-noise ratio, which in turn usually means doing it in the tooling.

In other words, Provenance and SBOM are operating at different levels of abstraction. Furthermore, Provenance can cover the trustworthiness of the SBOM, i.e. how the SBOM was generated.

**Comment by joshuagl** [comment]
> SBOM is intentionally not part of the Build track, though perhaps might be part of a future Vulnerabilities track. We really need a FAQ on this.

I've attempted to add a FAQ entry in https://github.com/slsa-framework/slsa/pull/827 based on Mark's quick take in https://github.com/slsa-framework/slsa/issues/691#issuecomment-1504083988

**Comment by MarkLodato** [comment]
Looking at this again after half a year of experience, what is the recommendation for where to put SBOMs, `byproducts` or `subject`? Does it matter how the SBOM was produced? Should the SBOM have its own, separate provenance?

- If the SBOM was generated by the user-defined build process, then I suppose `subject` is likely more appropriate since it was indeed an output.
- If the SBOM was generated by the build platform, I can see an argument either way.
  -  On the one hand, the provenance of the SBOM itself is valuable (esp. if that SBOM is not itself signed), so putting it in `subject` allows more uniform handling.
  - On the other hand, it's not a direct output of the build, so putting it in its own field might be appropriate.

Any thoughts? Overall I think I'd actually lean towards putting it in `subject` just for simplicity, but wanted to hear other opinions.

**Comment by laurentsimon** [comment]
putting in `subject` SGTM, and it's how most people do it today, afaik. It also provides a smooth upgrade path from signing to SLSA for the SBOM itself, as the very same tooling can be used during upgrade (verify_sig -> verify_slsa) for both build artifacts and SBOM artifacts.

**Comment by arewm** [comment]
While I don't have a preference either way, I suspect that this is hard to generalize.

If there is a task where the build platform generates the SBOM at the same time as the artifact itself, would it not be considered a direct output of the build? In this situation, even if the SBOM is not signed itself, if its immutable reference (i.e. OCI digest) is captured in the signed provenance, then trust can be extended to the SBOM content via the provenance.
